lib/middleware/verify_hideout_id.rb

```diff
@@ -0,0 +1,39 @@
+require 'json'
+
+require_relative '../../app/helpers/auth_helper.rb'
+
+module Middleware
+    class VerifyHideoutId
+      def initialize(app)
+        @app = app
+      end
+  
+      def call(env)
+        request = ActionDispatch::Request.new(env)
+        request_identifier = "#{request.method} #{request.path}"
+        if not ENV['PUBLIC_ROUTES'].include?(request_identifier)
+          authoriation_header = request.headers[:Authorization]
+          return [401, {}, []] if authoriation_header.nil?
+  
+          access_token = authoriation_header.split[1]
+          result = AuthHelper.validate_token_by_type(:ACCESS, access_token)
+          return [401, {}, []] unless result[:success]
+  
+          begin
+            payload = result[:payload]
+            payload.transform_keys!(&:to_sym)
+            current_hideout = Hideout.find_by(id: payload[:hideout_id])
```

Comment:
Retrieve the `hideout_id` from the `request` object. As I mentioned in my other comment, we won't have access to the hideout_id from the `payload` anymore. You'll need to use the `find_by!` function to throw an error if the hideout's not found.